scripts/globals/chocobo.lua

```diff
@@ -81,6 +81,38 @@ xi.chocobo.initZone = function(zone)
     end
 end
 
+xi.chocobo.renterOnTrade = function(player, npc, trade, eventSucceed, eventFail)
+    local zoneId = player:getZoneID()
+    local info   = chocoboInfo[zoneId]
+    if
+        trade:hasItemQty(xi.item.FREE_CHOCOPASS, 1) and
+        zoneId == (xi.zone.WINDURST_WOODS or xi.zone.BASTOK_MINES or xi.zone.SOUTHERN_SAN_DORIA) -- Chocopass does nothing for Non-Starter Cities
```

Comment:
This doesn't work. This is just `zoneId == xi.zone.WINDURST_WOODS`. Each zone needs to be checked individually. If you're pulling a commit from a different fork, you need to check if there were any changes/bug fixes applied to the code after the commit.